internal/docker/builder.go

```diff
@@ -292,32 +289,32 @@ func (d *Builder) construct(bprint b.Blueprint) (errs []error) {
 			continue
 		}
 		// collect and store access tokens as labels 'access_token_$userid: $token'
-		labels := make(map[string]string)
+		changes := make([]string, 0)
 		accessTokens := runner.AccessTokens(res.homeserver.Name)
 		if len(bprint.KeepAccessTokensForUsers) > 0 {
 			// only keep access tokens for specified users
 			for _, userID := range bprint.KeepAccessTokensForUsers {
 				tok, ok := accessTokens[userID]
 				if ok {
-					labels["access_token_"+userID] = tok
+					changes = appendLabelToChanges(changes, "access_token_"+userID, tok)
 				}
 			}
 		} else {
 			// keep all tokens
 			for k, v := range accessTokens {
-				labels["access_token_"+k] = v
+				changes = appendLabelToChanges(changes, "access_token_"+k, v)
 			}
 		}
 
 		deviceIDs := runner.DeviceIDs(res.homeserver.Name)
 		for userID, deviceID := range deviceIDs {
-			labels["device_id"+userID] = deviceID
+			changes = appendLabelToChanges(changes, "device_id"+userID, deviceID)
```

Comment:
This presumably is going to result in duplicate keys if there are two users with the same ID? Whereas previously we would just take the last value.

Reply:
Exact, I'll do what you suggested (use `labels` and then convert to `changes`) and this problem should be solved.